tracing/tracer.go

```diff
@@ -100,3 +103,17 @@ func (jr *jaegerReporter) UpdateGauge(name string, tags map[string]string, value
 func (jr *jaegerReporter) RecordTimer(name string, tags map[string]string, d time.Duration) {
 	jr.reporter.AllocateTimer(name, tags).ReportTimer(d)
 }
+
+// WithSpan is used for generating a span to be used in testing
+func WithSpan(t *testing.T, log ulog.Log, f func(opentracing.Span)) {
```

Comment:
+1. I feel like we probably need to avoid having testing stuffs expose through API (right now we have many of them) testutils sounds a right place to fit in